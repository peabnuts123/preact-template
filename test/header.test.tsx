import { h } from "preact";
import { render } from '@testing-library/preact';

import Header from "@app/components/header";

describe("Initial Test of the Header", () => {
  test("Header renders 3 nav items", () => {
    const { container } = render(<Header />);

    expect(container.querySelector('h1')?.textContent).toBe("Preact App");
    expect(container.querySelectorAll("a").length).toBe(3);
  });

  test('Matches snapshot', () => {
    const { baseElement } = render(<Header />);

    expect(baseElement).toMatchSnapshot();
  });
});
