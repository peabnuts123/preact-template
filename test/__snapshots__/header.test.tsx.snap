// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial Test of the Header Matches snapshot 1`] = `
<body>
  <div>
    <header
      class="header"
    >
      <h1>
        Preact App
      </h1>
      <nav>
        <a
          class="is-active"
          href="/"
        >
          Home
        </a>
        <a
          class=""
          href="/profile"
        >
          Me
        </a>
        <a
          class=""
          href="/profile/john"
        >
          John
        </a>
      </nav>
    </header>
  </div>
</body>
`;
